src/minerva_lib/crop.py

```diff
@@ -0,0 +1,298 @@
+
+    Returns:
+        Two-item int64 array i, j tile indices
+    '''
+    start = np.array(crop_origin)
+
+    return np.int64(np.floor(start / tile_size))
+
+
+def get_tile_count(tile_size, crop_origin, crop_size):
+    ''' Returns the number of tiles in the region.
+
+    Args:
+        tile_size: width, height of one tile
+        crop_origin: x, y origin of requested image region
+        crop_size: width, height of requested image region
+        image_size: width, height of full image
+
+    Returns:
+        Two-item int64 array i, j tile indices
+    '''
+    end = np.array(crop_origin) + crop_size
+
+    return np.int64(np.ceil(end / tile_size))
+
+
+def get_subregion(indices, tile_size, crop_origin, crop_size):
+    '''Determines the region within this specific tile
+    that is required for the requested image region.
+
+    Args:
+        indices: integer i, j tile indices
+        tile_size: width, height of one tile
+        crop_origin: x, y origin of requested image region
+        crop_size: width, height of requested image region
+
+    Returns:
+        The part of the tile within the requested region
+        Subregion start, end relative to origin of tile
+    '''
+
+    tile_start = np.int64(indices) * tile_size
+    crop_end = np.int64(crop_origin) + crop_size
+    tile_end = tile_start + tile_size
+
+    # Should be true that 0 <= start_uv < end_uv <= tile_size
+    start_uv = np.maximum(crop_origin, tile_start) - tile_start
+    end_uv = np.minimum(tile_end, crop_end) - tile_start
+
+    return [start_uv, end_uv]
+
+
+def get_position(indices, tile_size, crop_origin):
+    ''' Determines where in the requested image to insert
+    the required region from this specific tile.
+
+    Args:
+        indices: integer i, j tile indices
+        tile_size: width, height of one tile
+        crop_origin: x, y origin of requested image region
+
+    Returns:
+        Position of tile region within requested image region
+    '''
+
+    tile_start = np.int64(indices) * tile_size
+
+    # Should never be negative
+    return np.maximum(crop_origin, tile_start) - crop_origin
+
+
+def validate_region_bounds(crop_origin, crop_size, image_size):
+    ''' Decides if the image contains the requested image region.
+
+    Args:
+        crop_origin: x, y origin of requested image region
+        crop_size: width, height of requested image region
+        image_size: width, height of full image
+
+    Returns:
+        True if the requested image region is valid
+    '''
+
+    if any(np.less_equal(crop_size, 0)):
+        return False
+
+    if any(np.less(crop_origin, 0)):
+        return False
+
+    crop_end = np.array(crop_origin) + crop_size
+    if any(np.less(image_size, crop_end)):
+        return False
+
+    return True
+
+
+def select_tiles(tile_size, crop_origin, crop_size):
+    ''' Selects tile indices to fill the requested region.
+
+    Args:
+        tile_size: width, height of one tile
```

Comment:
It could be a tuple, list, or numpy array.  In the tests, I use a list. I can document and test these as any type you prefer. In the `blend` module, we use numpy arrays for arguments like color and range. For the `blend` module, I documented "float array" to emphasize that the "color" argument should have floating-point values, but it is true that the "color" argument could be a list, tuple, or array. In this new crop module, the documentation could explicitly ask for arrays of two integers, lists of two integers, or tuples of two integers. This is true of all the coordinate arguments in this module.